src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs

```diff
@@ -394,43 +392,38 @@ internal static HttpParseResult GetQuotedPairLength(string input, int startIndex
                 if (supportsNesting && (input[current] == openChar))
                 {
                     nestedCount++;
```

Comment:
Rather than incrementing and decrementing this, could nestedCount+1 just be passed as the argument to the recursive GetExpressionLength call?  If you did that, you'd also need to change the nestedCount > maxNestedCount to be >=, or pass 1 as the input from the wrapper methods instead of passing 0.